src/CattyTests/DataModel/VariablesContainerTest.swift

```diff
@@ -96,4 +96,145 @@ final class VariablesContainerTest: XCTestCase {
         XCTAssertEqual(1, container.allLists(for: objectA)?.count)
         XCTAssertEqual(1, container.allLists(for: objectB)?.count)
     }
+
+    func testAllVariables() {
+        let objectA = SpriteObject()
+        objectA.name = "testObjectA"
+
+        let userVariable1 = UserVariable()
+        userVariable1.name = "testName1"
+
+        let userVariable2 = UserVariable()
+        userVariable2.name = "testName2"
+
+        let container = VariablesContainer()
+
+        container.addObjectVariable(userVariable1, for: objectA)
+        var allVariable = container.allVariables() as! [UserVariable]
+
+        XCTAssertEqual(1, allVariable.count)
+        XCTAssertTrue(allVariable[0].name == userVariable1.name)
+
+        container.addObjectVariable(userVariable2, for: objectA)
+        allVariable = container.allVariables() as! [UserVariable]
+
+        XCTAssertEqual(2, allVariable.count)
```

Comment:
Please also add one list to document that this list will not be part of the result.